commands/fun/pokemon.js

```diff
@@ -0,0 +1,45 @@
+const Command = require(`${process.cwd()}/base/Command.js`);
+const { MessageEmbed } = require("discord.js");
+const pokemon = require("../../assets/json/pokemon.json");
+
+class Pokemon extends Command {
+  constructor(client) {
+    super(client, {
+      name: "pokemon",
+      description: "Guess That Pokemon",
+      usage: "pokemon",
+      aliases: ["guessthatpokemon"]
+    });
+  }
+
+  async run(message, args, level) { // eslint-disable-line no-unused-vars
+    const rand = Math.floor(Math.random() * 802);
+    const poke = rand > 0 ? rand : Math.floor(Math.random() * 802);
+    const pokem = pokemon[poke];
+    const embed = new MessageEmbed()
+      .setTitle("You have 15 seconds to guess ! Who's that Pokémon !")
+      .setAuthor(message.member.displayName, message.author.displayAvatarURL())
+      .setImage(pokem.imageURL)
+      .setColor(message.guild.me.roles.highest.color || 0x00AE86);
+    
+    const msg = await message.channel.send({ embed });
+    const filter = m => m.author.id === message.author.id;
+    const attempts = await msg.channel.awaitMessages(filter, { time: 15000, max: 1 });
+      
+    if (!attempts || !attempts.size) {
+      msg.delete();
+      return message.channel.send(`Ba-Baka! You took too long to answer. It was ${pokem.name}.`);
+    } 
+      
+    const answer = attempts.first().content.toLowerCase();  
+      
+    if (answer === pokem.name.toLowerCase()) {
```

Comment:
It's faster to check using RegExp#test here. ```javascript
new RegExp(poke.name, 'i').test(msg.content)
```

Reply:
@Soumil07 Lol no

Reply:
What is the problem with === smh, that is just over-complicating a simple thing.

Reply:
RegExp#test is no way near faster then `===`

Reply:
You're lowercasing twice which is why it's faster.

Reply:
no, lol

Reply:
RegExp#test is faster than lowercasing two different strings and check if they're equal as it checks char-by-char instead of converting all chars from both strings and then compare.

The good thing of RegExp#test when well used is that if you use a pattern that only matches if the string starts with something (`^`), it'll return false as soon as one char doesn't match in insensitive case to the another.

The downside is the loss of userfriendlyness and increasement of CPU and RAM usage for a simple task, you're better caching that RegExp (and escaping it so you don't get [ReDoS](https://en.wikipedia.org/wiki/ReDoS)). This is not the case, so lowercasing two strings for a single use is way better.